src/object/cli_obj.c

```diff
@@ -6096,6 +6096,9 @@ dc_obj_query_key(tse_task_t *api_task)
 
 			leader = obj_grp_leader_get(obj, i, map_ver, NIL_BITMAP);
 			if (leader >= 0) {
+				if (obj_is_ec(obj) && !is_ec_parity_shard(leader, obj_get_oca(obj)))
+					goto non_leader;
+
```

Comment:
thanks for fixing this. Hmm, this probably need moves to obj_grp_leader_get(), which probably needs a bit cleanup. sorry. Thanks.

Reply:
no problem, we can cleanup in following patch. thx